core/drivers/crypto/caam/include/caam_common.h

```diff
@@ -0,0 +1,103 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   CAAM driver common include file.
+ *         Definition of the internal driver status codes.
+ */
+
+#ifndef __CAAM_COMMON_H__
+#define __CAAM_COMMON_H__
+
+#include <caam_desc_helper.h>
+#include <caam_status.h>
+#include <caam_trace.h>
+#include <types_ext.h>
+
+/*
+ * Definition of the number of CAAM Jobs to manage in JR queues
+ */
+#define NB_JOBS_QUEUE 10
+
+/*
+ * Flag Job Ring Owner is Secure
+ */
+#define JROWNER_SECURE 0x10
+
+/*
+ * Job Ring Owner. Enumerate Id (expect the Secure Flag) correspond
+ * to the HW ID.
+ */
+#if !defined(CFG_MX7ULP)
+enum caam_jr_owner {
+	JROWN_ARM_NS = 0x1,                  /* Non-Secure ARM */
+	JROWN_ARM_S  = JROWNER_SECURE | 0x1, /* Secure ARM */
+};
+#else
+enum caam_jr_owner {
+	JROWN_ARM_NS = 0x4,                  /* Non-Secure ARM */
+	JROWN_ARM_S = JROWNER_SECURE | 0x4,  /* Secure ARM */
+};
+#endif
+
+/*
+ * Definition of a CAAM buffer type
+ */
+struct caambuf {
+	uint8_t *data;   /* Data buffer */
+	paddr_t paddr;   /* Physical address of the buffer */
+	size_t length;   /* Number of bytes in the data buffer */
+	uint8_t nocache; /* =1 if buffer is not cacheable */
```

Comment:
Negative logic usually make it harder to understand the code. I'd prefer `bool cached`. Keep it as it is if you think it's important.

Reply:
Yes but I'll keep it as is for the simple reason that be default with the variable initialization at 0's the default `nocache` is as expected without adding a new code line to set the value.